src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c

```diff
@@ -208,88 +216,129 @@ static inline void reg_wr(void __iomem *reg, u32 val)
 }
 
 static int icap_cache_bitstream_axlf_section(struct platform_device *pdev,
-	const struct axlf *xclbin, enum axlf_section_kind kind);
+	const struct axlf *xclbin, enum axlf_section_kind kind, uint32_t slot_id);
 static void icap_set_data(struct icap *icap, struct xcl_pr_region *hwicap);
 static uint64_t icap_get_data_nolock(struct platform_device *pdev, enum data_kind kind);
 static uint64_t icap_get_data(struct platform_device *pdev, enum data_kind kind);
 static void icap_refresh_addrs(struct platform_device *pdev);
-static int icap_calib_and_check(struct platform_device *pdev);
+static int icap_calib_and_check(struct platform_device *pdev, uint32_t slot_id);
 static void icap_probe_urpdev(struct platform_device *pdev, struct axlf *xclbin,
-	int *num_urpdev, struct xocl_subdev **urpdevs);
+	int *num_urpdev, struct xocl_subdev **urpdevs, uint32_t slot_id);
+
+static struct islot_info *
+icap_slot_init(struct icap *icap, uint32_t slot_id)
+{
+	struct islot_info *islot = icap->slot_info[slot_id];
+	
+	if (islot) {
+		vfree(islot);
```

Comment:
you freed the pointer but left a dangling one in icap->slot_info[slot_id]?

Reply:
Fixed.